libs/libdaemon/include/daemon/BaseDaemon.h

```diff
@@ -142,6 +144,9 @@ class BaseDaemon : public Poco::Util::ServerApplication
 	/// thread safe
 	virtual void onInterruptSignals(int signal_id);
 
+	template <class Daemon>
+	static boost::optional<Daemon &> tryGetInstance();
```

Comment:
We are using `std::experimental::optional` instead of `boost::optional` everywhere. There is no practical difference, but better to be consistent.